### PR TITLE
[MOS-812] Improved CUSD message handling

### PR DESCRIPTION
**Pull request by marziel**
* Added handling for shorter +CUSD formats, thus got rid of a crash.
* Added handling for the "push session"+"action required" state of a USSD interaction with the network.

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible
- [x] Has changelog entry added

### CODE REVIEW

File: module-cellular/test/unittest_URC.cpp [hunk 1]
```diff
@@ -279,78 +279,6 @@ TEST_CASE("+Qind: FOTA")
     }
 }
 
-TEST_CASE("+Cusd")
```
Comment (SP2FET) [inline module-cellular/test/unittest_URC.cpp:282]: maybe it's worth adding some test cases for the new implementation? especially that you're using regex :P

Comment (marziel) [reply module-cellular/test/unittest_URC.cpp:282]: Yeah, I'm already working on them...

Comment (marziel) [reply module-cellular/test/unittest_URC.cpp:282]: Decided to temporarily withdraw the UTs, see https://appnroll.atlassian.net/browse/MOS-812?focusedCommentId=69817.

File: module-cellular/at/UrcCusd.hpp [hunk 2]
```diff
@@ -30,26 +32,17 @@ namespace at::urc
...
         static constexpr std::string_view head = "+CUSD";
-        static auto isURC(const std::string &uHead) -> bool
-        {
-            return uHead.find(Cusd::head) != std::string::npos;
-        }
 
-        using Urc::Urc;
+        explicit Cusd(const std::string &urcBody, const std::string &urcHead = {});
 
+        static auto isURC(const std::string &uHead) -> bool;
```
Comment (rrandomsky) [inline module-cellular/at/UrcCusd.hpp:40]: Sugestion:
make method  `const`
`static auto isURC(const std::string &uHead) const -> bool;`

Comment (marziel) [reply module-cellular/at/UrcCusd.hpp:40]: A `static` method can't be `const` at the same time as it isn't bound to any instance of this class.

**Review by kkleczkowski — APPROVED** [review]

File: module-services/service-cellular/call/tests/CMakeLists.txt [hunk 3]
```diff
@@ -10,6 +10,7 @@ add_catch2_executable(
         sml::sml
         sml::utils::logger
         test::fakeit
+	libphonenumber
```
Comment (marziel) [inline module-services/service-cellular/call/tests/CMakeLists.txt:13]: Added to fix linkage problems caused by messy dependencies

**Pull request merged**